R/dparse.R

```diff
@@ -0,0 +1,48 @@
+#' @title String parses
+#' @description Parse a custom string that represents an R6 distribution
+#' @details Transform a custom (user) input to a R6 object.
+#'
+#' This function is specially useful when you expect a user input which should
+#' result in specific distribution. The distribution name must be the
+#' ShortName, ClassName or Alias listed in the package. You can list them
+#' using the \link{listDistributions} function.
+#'
+#' Text call must be in the following format: Distribution(\[params\]), e.g.
+#' T() or Normal(mean = 3).
+#'
+#' The parameters can be entered in any order, but they must be named. If no
+#' parameter is given, the default parametrization provided in $new() call
+#' is used.
+#'
+#' @param toparse character
+#'
+#' @return Returns an R6 distribution object
+#'
+#' @examples
+#' dparse("N()")
+#' dparse("T(df = 4)")
+#' dparse("chisq(df = 3)")
+#' @export
+dparse <- function(toparse) {
+  # Check correct format
+  match <- grepl("^[a-zA-Z]+\\(.*\\)$", toparse)
+  if (!match) stop(sprintf("Call '%s' is not valid. The required structure is: Distributino([params]), e.g. T() or Normal(mean = 3).",
+                           toparse))
```

Comment:
I'd actually suggest removing the check here as you only need the 'Distribution not found' check you have below and then distr6 will handle parameters internally

Reply:
I played around with a version that did not have this check, but it was quite difficult using only base R to catch error several error calls, e.g., norm or norm) because the distribution name and parameters were then mixed up in strange ways. I decided to keep this check to make things simple as possible - any call that passes it matches the required format defined in the description. I hope it's not too much trouble.

Reply:
Makes sense, thanks!